src/pyodide/internal/workers.py

```diff
@@ -149,54 +157,202 @@ def json(
+            map(lambda x: self._process_value_for_py(x), self._js_form_data.getAll(key))
+        )
+
+    @property
+    def native(self) -> "js.FormData":
+        return self._js_form_data
+
+
+BlobValue = (
+    "str | bytes | js.ArrayBuffer | js.TypedArray | js.DataView | js.Blob | Blob"
+)
+
+
+class BlobEnding(StrEnum):
+    TRANSPARENT = "transparent"
+    NATIVE = "native"
+
+
+class BlobKwargs(TypedDict, total=False):
+    content_type: str
+    endings: BlobEnding
+
+
+class Blob:
+    def __init__(
+        self,
+        blob_parts: "Iterable[BlobValue] | BlobValue",
+        **options: Unpack[BlobKwargs],
+    ):
+        if "content_type" in options:
+            options["type"] = options["content_type"]
+        if "endings" in options:
+            options["endings"] = str(options["endings"])
+
+        is_single_item = not hasattr(blob_parts, "__iter__")
+        if is_single_item:
+            if "type" not in options:
+                # Inherit the `type` if we have a single item.
+                self._js_blob = (
+                    blob_parts.native if isinstance(blob_parts, Blob) else blob_parts
+                )
+                return
+
+            # Otherwise create a new Blob below.
+            blob_parts = [blob_parts]
+
+        with ExitStack() as stack:
+            args = [stack.enter_context(self._make_blob_entry(e)) for e in blob_parts]
+            self._js_blob = js.Blob.new(to_js(args), **options)
+
+    @property
+    def size(self) -> int:
+        return self._js_blob.size
+
+    @property
+    def type(self) -> str:
+        return self._js_blob.type
+
+    @property
+    def name(self) -> str:
+        if self._js_blob.constructor.name != "File":
+            raise TypeError(
+                "Underlying Blob type is not a File and so has no file name."
+            )
+        return self._js_blob.name
+
+    @property
+    def last_modified(self) -> int:
+        if self._js_blob.constructor.name != "File":
+            raise TypeError(
+                "Underlying Blob type is not a File and so has no last modified "
+                + "timestamp."
+            )
+        return self._js_blob.last_modified
+
+    @property
+    def native(self) -> "js.Blob":
+        return self._js_blob
+
+    async def text(self) -> str:
+        return await self.native.text()
+
+    async def bytes(self) -> bytes:
+        return (await self.native.arrayBuffer()).to_bytes()
+
+    def slice(
+        self,
+        start: int | None = None,
+        end: int | None = None,
+        content_type: str | None = None,
+    ):
+        return self.native.slice(start, end, content_type)
+
+    def _supports_buffer_protocol(self, o):
+        try:
+            # memoryview used only for testing type; 'with' releases the view instantly
+            with memoryview(o):
+                return True
+        except TypeError:
+            return False
+
+    @contextmanager
+    def _make_blob_entry(self, e):
```

Comment:
These should be top level functions, not methods. Or at least make them static.

Reply:
ahh, I knew I was forgetting something :)

Done.